### ISSUE TITLE
Casting a seq to another seq generates invalid code with --newruntime

### ISSUE THREAD

**Issue by Clyybber** [nim-lang/Nim#11018]
Casting a seq or a string to another seq generates invalid C code with --newruntime

### A couple of examples:
seq -> seq

```nim
discard cast[seq[uint8]](@[1])
```
string -> seq
```nim
discard cast[seq[uint8]]("test")
```

### Current Output
```
Error: execution of an external compiler program 'gcc -c  -w  -I/home/clyybber/builds/nim/lib -I/home/clyybber/projects/tests -o /home/clyybber/.cache/nim/convrt_d/convrt.c.o /home/clyybber/.cache/nim/convrt_d/convrt.c' failed with exit code: 1

/home/clyybber/.cache/nim/convrt_d/convrt.c: In function ‘NimMainModule’:
/home/clyybber/.cache/nim/convrt_d/convrt.c:157:2: error: conversion to non-scalar type requested
  (void)(((tySequence_6H5Oh5UUvVCLiakt9aTwtUQ) (TM_KUSKOxFeGyNT8JCQmxu3cw_2)));
  ^
```

### Expected Output

Compile successfully

### Additional Information
Casting an array to a seq does not generate invalid code:
```nim
discard cast[seq[uint8]]([1])
```

```
$ nim -v
Nim Compiler Version 0.19.9
```

**Comment by krux02** [comment]
I just tried to reproduce the bug, these are my results:

```nim
echo cast[seq[uint8]](@[1,2,3,4,5,6,7,8,9])   # @[1, 0, 0, 0, 0, 0, 0, 0, 2]
echo cast[seq[uint8]]("test")                 # @[116, 101, 115, 116]
echo cast[seq[uint8]]([1])                    # segfault (as expected)
```

What you are doing is very risky, it exploits similarities in the implementation of seq and string, even though the implementation of seq and string is not specified. It is allowed to be changed in future versions of Nim.

**Comment by Clyybber** [comment]
@krux02 I'm aware that this is missing setLen and is risky, but this is not about running the program successfully, but compiling it.
You probably missed the --newruntime switch when you compiled your program.

**Comment by krux02** [comment]
yes you are right, I did miss the newruntime switch

**Comment by Clyybber** [comment]
@krux02 Yeah, I'm aware. When doing it the other way around, 
```nim
someSeq.setLen((someSeq.len * sizeof(originalType)/sizeof(TypeWeCastedTo))).ceil.int)
``` 
or 
```nim
someSeq.setLen((originalSeq.len / sizeof(TypeWeCastedTo)/sizeof(originalType))).ceil.int)
```
 is required.
Interestingly your previous code sample using `echo` instead of `discard` revealed a big problem in my attempt to fix this issue in #11019 , that is, it's missing the len of the seq as it only casts the contents of the seq, so the first snippet would no longer work.
Also this makes `$` no longer work on the casted seq, as it has no length :P
